### ISSUE TITLE
`[RUF055]` False positive for `re.split` with empty string pattern

### ISSUE THREAD

**Issue by harupy** [astral-sh/ruff#23629]
### Summary

`re.split("", s)` is flagged by RUF055 and fixed to `s.split("")`, but `str.split` raises `ValueError: empty separator` for empty separators while `re.split` succeeds:

```python
import re

s = "abc"
re.split("", s)   # ['', 'a', 'b', 'c', '']
s.split("")        # ValueError: empty separator
```

### Version

f1a9a703e1992520d7edd3d907d2ed6a019a120b

### Potential fix

Add a guard to skip if the separator is an empty string.

**Comment by ntBre** [comment]
Thank you!
